lib/ansible/plugins/action/network.py

```diff
@@ -60,7 +60,13 @@ def _handle_backup_option(self, result, hostname, backup):
             # User requested backup and no error occurred in module.
             # NOTE: If there is a parameter error, _backup key may not be in results.
             filepath = self._write_backup(hostname, backup)
+            tmp = filepath[::-1].split('.', 1)
+
             result['backup_path'] = filepath
+            result['filename'] = filepath.split('/')[-1]
+            result['shortname'] = tmp[1][::-1]
+            result['date'] = tmp[0][::-1].split('@')[0]
```

Comment:
If `date` and `time` is not added as part of a file name (in case of a configurable backup path option) `date` and `time` logic can be changed to fetch from the file information probably using os.stat()

Reply:
Sounds good. Added a new commit for it.